app/src/main/java/net/kwatts/powtools/MainActivity.java

```diff
@@ -480,11 +432,18 @@ public void onError(Throwable e) {
                 Intent i = new Intent(this, MainPreferencesActivity.class);
                 startActivity(i);
                 break;
+            case R.id.menu_refresh:
+                createNewRide();
```

Comment:
One issue that has always bugged me is the ride or session "flow" that marks the full ride. Right now it a ride starts when BLE connects to the OW, then - and this is the dirty - ends the ride when the BLE disconnects. So a full 30 minute ride that should be just one ride is instead broken into 2 or three - and you can see this w/ the multiple CSV files being created - because BLE disconnects and is so flaky. The official OW app deals w/ this by having you manually start the ride then manually end the ride. That's not easy and a bit of a pain, I like kicking off the ride session when first connecting, but we should think about doing some sort of "ride end" button or some state other than BLE disconnects.

Reply:
Hrmmm, I didn't realize this was happening. I wonder if you wouldn't like this idea better: come up with a definition of 'ride' that includes what separates a ride. So if we have two moments that are 5 minutes apart, then start a new ride, otherwise treat it as the same ride. Also could allow a user to manually stop & start-new if you tell me where that button would work.

Reply:
Ahh ya I think you got it. We can add the start/stop to the toolbar. For the gap moment time duration between rides, my suggestion is we make that an optional preference that can be set in minutes. Default to something low (30 seconds/1 minute). Otherwise it could be annoying waiting 5 minutes after a ride until you can see the ride session deets.

Reply:
nice thing about DB is you wouldn't have to wait to see it. In fact, (if you could figure out how to make it look good) it would be cool
to allow switching between speedometer and graph!